### ISSUE TITLE
Buildx does not allow oci-layout --context to be readonly.

### ISSUE THREAD

**Issue by thesayyn** [docker/buildx#2753]
### Contributing guidelines

- [x] I've read the [contributing guidelines](https://github.com/docker/buildx/blob/master/.github/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [x] ... the documentation does not mention anything about my problem
- [x] ... there are no open or closed issues that are related to my problem

### Description

Currently when attaching an oci-layout context to buildx, it wants to create some lock/ingest files within the target directory even though those files are not used in an meaningful way  in this instance as far as in can see.

Here it wants to create `index.json.lock` if the digest is not provided

https://github.com/docker/buildx/blob/746eadd16e461ac1a961bb22e7fea8f1d05da88f/build/opt.go#L509

And here if you provide a tag. 

https://github.com/docker/buildx/blob/746eadd16e461ac1a961bb22e7fea8f1d05da88f/build/opt.go#L514

Context: 

I am trying to build a Bazel ruleset to allow running buildx under Bazel, and to attach contexts to buildx so it can be copied into the resulting container,  however, as a accidental write prevention measure, Bazel marks artifacts from other build rules as readonly which is where buildx fails because it assumes the stores to be writeable even though it does not need them to be writable in this instance.

### Expected behaviour

I'd expect the oci-layout store to not create these ingestion/lock files eagerly. 

### Actual behaviour

oci-layout store creates ingest/lock files eagerly preventing the directories to be marked readonly. 

### Buildx version

v0.17.1

### Builders list

```
NAME/NODE         DRIVER/ENDPOINT             STATUS  PLATFORMS
builder-docker    docker-container                    
  builder-docker0 unix:///var/run/docker.sock running linux/amd64, linux/amd64/v2, linux/amd64/v3, linux/amd64/v4, linux/386
default *         docker                              
  default         default                     running linux/amd64, linux/386
```

### Configuration

```dockerfile
FROM ghcr.io/kamilkisiela/graphql-hive/apollo-router:f73977ea2219fb16a77602fd7e7c79069f1447fa

SHELL ["/bin/bash", "-c"]

RUN apt-get -y update
RUN apt-get install --no-install-recommends -y jq
RUN apt-get clean
RUN rm -rf /var/lib/apt/lists/*

```

### Build logs
```
buildx build . --file services/applications/apollo-router/Dockerfile --builder builder-docker '--output=type=oci,tar=false,dest=bazel-out/k8-fastbuild/bin/services/applications/apollo-router/build' '--build-context=ghcr.io/kamilkisiela/graphql-hive/apollo-router:f73977ea2219fb16a77602fd7e7c79069f1447fa=oci-layout://bazel-out/k8-fastbuild/bin/external/rules_oci~~oci~o_apollo-router_linux_amd64/layout@sha256:7554a215f5d770c12bc2bf3486b48430ee867209a9e1ae146c62121eb6f93d35')

ERROR: invalid store at bazel-out/k8-fastbuild/bin/external/rules_oci~~oci~o_apollo-router_linux_amd64/layout: mkdir bazel-out/k8-fastbuild/bin/external/rules_oci~~oci~o_apollo-router_linux_amd64/layout/ingest: permission denied

```

### Additional info

FWIW, i filed this issue in containerd repo https://github.com/containerd/containerd/issues/10885 as one of the offending library seems to be coming from there.

**Comment by thesayyn** [comment]
> then the workaround is to just make sure that an empty ingest directory already exists in the build-context directory.

i see that could be useful as a workaround, makes sense to create that folder when creating the oci-layout, i'll look into that but it would be great if this was fixed here.


> Maybe you are on a system that doesn't support file locks

oh the reason i don't get the `index.json.lock` issue is because i use a digest (to workaround) so that codepath is never hit. right here https://github.com/docker/buildx/blob/746eadd16e461ac1a961bb22e7fea8f1d05da88f/build/opt.go#L508

**Comment by tonistiigi** [comment]
PRs: https://github.com/moby/buildkit/pull/5452 https://github.com/containerd/containerd/pull/10894

**Comment by thesayyn** [comment]
Thank you @tonistiigi, i can confirm that creating `ingest` and `index.json.lock` works as a workaround.

**Comment by tonistiigi** [comment]
The lock side of this has been fixed in buildx v0.18.0 , the "ingest" side will need to wait for containerd libs update.

**Comment by thompson-shaun** [comment]
backport to containerd 1.7.x can be tracked at https://github.com/containerd/containerd/pull/10913
